Bottom: deae97595d94da9115144af3d845d117506eff13
Top:    be1dc7d79cd27a817e87bb44aa79287e7fbd68ad
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-24 15:38:12 +0000

Remove *TypePtr and *TypeHandle aliases now we no longer use them

(May break build of plugins, hence keeping this as a separately
revertable commit).


---

diff --git a/src/global.h b/src/global.h
index 4b21931..7b28396 100644
--- a/src/global.h
+++ b/src/global.h
@@ -118,7 +118,7 @@ typedef struct
 {
   unsigned long f;		/* generic flags */
   unsigned char t[(MAX_LAYER + 1) / 2];	/* thermals */
-} FlagType, *FlagTypePtr;
+} FlagType;
 
 #ifndef __GNUC__
 #define __FUNCTION1(a,b) a ":" #b
@@ -190,7 +190,7 @@ typedef struct			/* holds information about output window */
     fgGC,			/* changed from some routines */
     pmGC;			/* depth 1 pixmap GC to store clip */
 }
-OutputType, *OutputTypePtr;
+OutputType;
 
 /* ----------------------------------------------------------------------
  * layer group. A layer group identifies layers which are always switched
@@ -200,7 +200,7 @@ typedef struct
 {
   Cardinal Number[MAX_LAYER],	/* number of entries per groups */
     Entries[MAX_LAYER][MAX_LAYER + 2];
-} LayerGroupType, *LayerGroupTypePtr;
+} LayerGroupType;
 
 struct BoxType		/* a bounding box */
 {
@@ -212,13 +212,13 @@ typedef struct
 {
   Coord x, y;
   Coord width, height;
-} RectangleType, *RectangleTypePtr;
+} RectangleType;
 
 typedef struct
 {
   char *name;
   char *value;
-} AttributeType, *AttributeTypePtr;
+} AttributeType;
 
 struct AttributeListType
 {
@@ -234,24 +234,24 @@ struct AttributeListType
    based on this. */
 typedef struct {
   ANYOBJECTFIELDS;
-} AnyObjectType, *AnyObjectTypePtr;
+} AnyObjectType;
 
 typedef struct			/* a line/polygon point */
 {
   Coord X, Y, X2, Y2;	/* so Point type can be cast as BoxType */
   long int ID;
-} PointType, *PointTypePtr;
+} PointType;
 
 /* Lines, rats, pads, etc.  */
 typedef struct {
   ANYLINEFIELDS;
-} AnyLineObjectType, *AnyLineObjectTypePtr;
+} AnyLineObjectType;
 
 typedef struct			/* holds information about one line */
 {
   ANYLINEFIELDS;
   char *Number;
-} LineType, *LineTypePtr;
+} LineType;
 
 typedef struct
 {
@@ -261,7 +261,7 @@ typedef struct
   BYTE Direction;
   char *TextString;		/* string */
   void *Element;
-} TextType, *TextTypePtr;
+} TextType;
 
 struct polygon_st			/* holds information about a polygon */
 {
@@ -287,7 +287,7 @@ typedef struct			/* holds information about arcs */
   Coord Width, Height,		/* length of axis */
     X, Y;			/* center coordinates */
   Angle StartAngle, Delta;	/* the two limiting angles in degrees */
-} ArcType, *ArcTypePtr;
+} ArcType;
 
 struct rtree
 {
@@ -313,13 +313,13 @@ typedef struct			/* holds information about one layer */
   AttributeListType Attributes;
   int no_drc; /* whether to ignore the layer when checking the design rules */
 }
-LayerType, *LayerTypePtr;
+LayerType;
 
 typedef struct			/* a rat-line */
 {
   ANYLINEFIELDS;
   Cardinal group1, group2;	/* the layer group each point is on */
-} RatType, *RatTypePtr;
+} RatType;
 
 struct pad_st			/* a SMD pad */
 {
@@ -365,7 +365,7 @@ typedef struct
   GList *Arc;
   BoxType VBox;
   AttributeListType Attributes;
-} ElementType, *ElementTypePtr, **ElementTypeHandle;
+} ElementType;
 
 /* ---------------------------------------------------------------------------
  * symbol and font related stuff
@@ -378,7 +378,7 @@ typedef struct			/* a single symbol */
     LineMax;
   Coord Width, Height,		/* size of cell */
     Delta;			/* distance to next symbol */
-} SymbolType, *SymbolTypePtr;
+} SymbolType;
 
 typedef struct			/* complete set of symbols */
 {
@@ -387,7 +387,7 @@ typedef struct			/* complete set of symbols */
   BoxType DefaultSymbol;	/* the default symbol is a filled box */
   SymbolType Symbol[MAX_FONTPOSITION + 1];
   bool Valid;
-} FontType, *FontTypePtr;
+} FontType;
 
 typedef struct			/* holds all objects */
 {
@@ -403,7 +403,7 @@ typedef struct			/* holds all objects */
   struct PCBType *pcb;
   LayerType Layer[MAX_LAYER + 2];	/* add 2 silkscreen layers */
   int polyClip;
-} DataType, *DataTypePtr;
+} DataType;
 
 typedef struct			/* holds drill information */
 {
@@ -417,14 +417,14 @@ typedef struct			/* holds drill information */
     PinMax;			/* max number of coordinates from malloc() */
   PinType **Pin;		/* coordinates to drill */
   ElementType **Element;	/* a pointer to an array of element pointers */
-} DrillType, *DrillTypePtr;
+} DrillType;
 
 typedef struct			/* holds a range of Drill Infos */
 {
   Cardinal DrillN,		/* number of drill sizes */
     DrillMax;			/* max number from malloc() */
   DrillType *Drill;		/* plated holes */
-} DrillInfoType, *DrillInfoTypePtr;
+} DrillInfoType;
 
 typedef struct
 {
@@ -434,7 +434,7 @@ typedef struct
     Keepaway;			/* min. separation from other nets */
   char *Name;
   int index;
-} RouteStyleType, *RouteStyleTypePtr;
+} RouteStyleType;
 
 /* ---------------------------------------------------------------------------
  * structure used by library routines
@@ -448,8 +448,7 @@ typedef struct
    *Package,			/* package */
    *Value,			/* the value field */
    *Description;		/* some descritional text */
-} LibraryEntryType, *LibraryEntryTypePtr;
-//typedef LibraryEntryType *LibraryEntryTypePtr;
+} LibraryEntryType;
 
 /* If the internal flag is set, the only field that is valid is Name,
    and the struct is allocated with malloc instead of
@@ -467,14 +466,14 @@ typedef struct
   char flag;			/* used by the netlist window to enable/disable nets */
   char internal;		/* if set, this is an internal-only entry, not
 				   part of the global netlist. */
-} LibraryMenuType, *LibraryMenuTypePtr;
+} LibraryMenuType;
 
 typedef struct
 {
   Cardinal MenuN;               /* number of objects */
   Cardinal MenuMax;             /* number of reserved memory locations */
   LibraryMenuType *Menu;      /* the entries */
-} LibraryType, *LibraryTypePtr;
+} LibraryType;
 
 
   /* The PCBType struct holds information about board layout most of which is
@@ -528,7 +527,7 @@ typedef struct PCBType
 
   bool is_footprint;		/* If set, the user has loaded a footprint, not a pcb. */
 }
-PCBType, *PCBTypePtr;
+PCBType;
 
 typedef struct			/* information about the paste buffer */
 {
@@ -536,7 +535,7 @@ typedef struct			/* information about the paste buffer */
   BoxType BoundingBox;
   DataType *Data;		/* data; not all members of PCBType */
   /* are used */
-} BufferType, *BufferTypePtr;
+} BufferType;
 
 /* ---------------------------------------------------------------------------
  * some types for cursor drawing, setting of block and lines
@@ -547,7 +546,7 @@ typedef struct			/* rubberband lines for element moves */
   LayerType *Layer;		/* layer that holds the line */
   LineType *Line;		/* the line itself */
   PointType *MovedPoint;	/* and finally the point */
-} RubberbandType, *RubberbandTypePtr;
+} RubberbandType;
 
 typedef struct			/* current marked line */
 {
@@ -555,7 +554,7 @@ typedef struct			/* current marked line */
     Point2;
   long int State;
   bool draw;
-} AttachedLineType, *AttachedLineTypePtr;
+} AttachedLineType;
 
 typedef struct			/* currently marked block */
 {
@@ -563,7 +562,7 @@ typedef struct			/* currently marked block */
     Point2;
   long int State;
   bool otherway;
-} AttachedBoxType, *AttachedBoxTypePtr;
+} AttachedBoxType;
 
 typedef struct			/* currently attached object */
 {
@@ -577,7 +576,7 @@ typedef struct			/* currently attached object */
   Cardinal RubberbandN,		/* number of lines in array */
     RubberbandMax;
   RubberbandType *Rubberband;
-} AttachedObjectType, *AttachedObjectTypePtr;
+} AttachedObjectType;
 
 enum crosshair_shape
 {
@@ -599,13 +598,13 @@ typedef struct			/* holds cursor information */
   PolygonType AttachedPolygon;
   AttachedObjectType AttachedObject;	/* data of attached objects */
   enum crosshair_shape shape; 	/* shape of crosshair */
-} CrosshairType, *CrosshairTypePtr;
+} CrosshairType;
 
 typedef struct
 {
   bool status;
   Coord X, Y;
-} MarkType, *MarkTypePtr;
+} MarkType;
 
 /* ---------------------------------------------------------------------------
  * our resources
@@ -696,7 +695,7 @@ typedef struct			/* some resources... */
     AutoPlace;			/* flag which says we should force placement of the
 				   windows on startup */
 }
-SettingType, *SettingTypePtr;
+SettingType;
 
 /* ----------------------------------------------------------------------
  * pointer to low-level copy, move and rotate functions
@@ -715,7 +714,7 @@ typedef struct
   void *(*Point) (LayerType *, PolygonType *, PointType *);
   void *(*Arc) (LayerType *, ArcType *);
   void *(*Rat) (RatType *);
-} ObjectFunctionType, *ObjectFunctionTypePtr;
+} ObjectFunctionType;
 
 /* ---------------------------------------------------------------------------
  * structure used by device drivers
@@ -728,7 +727,7 @@ typedef struct			/* holds a connection */
   void *ptr1, *ptr2;		/* the object of the connection */
   Cardinal group;		/* the layer group of the connection */
   LibraryMenuType *menu;	/* the netmenu this *SHOULD* belong too */
-} ConnectionType, *ConnectionTypePtr;
+} ConnectionType;
 
 typedef struct			/* holds a net of connections */
 {
@@ -736,28 +735,28 @@ typedef struct			/* holds a net of connections */
     ConnectionMax;		/* max connections from malloc */
   ConnectionType *Connection;
   RouteStyleType *Style;
-} NetType, *NetTypePtr;
+} NetType;
 
 typedef struct			/* holds a list of nets */
 {
   Cardinal NetN,		/* the number of subnets contained */
     NetMax;			/* max subnets from malloc */
   NetType *Net;
-} NetListType, *NetListTypePtr;
+} NetListType;
 
 typedef struct			/* holds a list of net lists */
 {
   Cardinal NetListN,		/* the number of net lists contained */
     NetListMax;			/* max net lists from malloc */
   NetListType *NetList;
-} NetListListType, *NetListListTypePtr;
+} NetListListType;
 
 typedef struct			/* holds a generic list of pointers */
 {
   Cardinal PtrN,		/* the number of pointers contained */
     PtrMax;			/* max subnets from malloc */
   void **Ptr;
-} PointerListType, *PointerListTypePtr;
+} PointerListType;
 
 typedef struct
 {
@@ -765,7 +764,7 @@ typedef struct
     BoxMax;			/* max boxes from malloc */
   BoxType *Box;
 
-} BoxListType, *BoxListTypePtr;
+} BoxListType;
 
 struct drc_violation_st
 {
diff --git a/src/mymem.h b/src/mymem.h
index 71c0a84..46061a3 100644
--- a/src/mymem.h
+++ b/src/mymem.h
@@ -62,7 +62,7 @@ typedef struct
 {
   size_t MaxLength;
   char *Data;
-} DynamicStringType, *DynamicStringTypePtr;
+} DynamicStringType;
 
 RubberbandType * GetRubberbandMemory (void);
 PinType * GetPinMemory (ElementType *);
